components/console-backend-service/internal/domain/ui/clustermicrofrontend_converter.go

```diff
@@ -0,0 +1,79 @@
+package ui
+
+import (
+	uiV1alpha1v "github.com/kyma-project/kyma/common/microfrontend-client/pkg/apis/ui/v1alpha1"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+)
+
+type clusterMicrofrontendConverter struct{}
+
+func (c *clusterMicrofrontendConverter) ToGQL(in *uiV1alpha1v.ClusterMicroFrontend) (*gqlschema.ClusterMicrofrontend, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	navigationNodes, err := c.NavigationNodesToGQLs(in.Spec.CommonMicroFrontendSpec.NavigationNodes)
+	if err != nil {
+		return nil, err
+	}
+
+	mf := gqlschema.ClusterMicrofrontend{
+		Name:            in.Name,
+		Placement:       in.Spec.Placement,
+		Version:         in.Spec.CommonMicroFrontendSpec.Version,
+		Category:        in.Spec.CommonMicroFrontendSpec.Category,
+		ViewBaseURL:     in.Spec.CommonMicroFrontendSpec.ViewBaseURL,
+		NavigationNodes: navigationNodes,
+	}
+
+	return &mf, nil
+}
+
+func (c *clusterMicrofrontendConverter) ToGQLs(in []*uiV1alpha1v.ClusterMicroFrontend) ([]gqlschema.ClusterMicrofrontend, error) {
+	var result []gqlschema.ClusterMicrofrontend
+	for _, u := range in {
+		converted, err := c.ToGQL(u)
+		if err != nil {
+			return nil, err
+		}
+
+		if converted != nil {
+			result = append(result, *converted)
+		}
+	}
+	return result, nil
+}
+
+func (c *clusterMicrofrontendConverter) NavigationNodeToGQL(in *uiV1alpha1v.NavigationNode) (*gqlschema.NavigationNode, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	navigationNode := gqlschema.NavigationNode{
+		Label:            in.Label,
+		NavigationPath:   in.NavigationPath,
+		ViewURL:          in.ViewURL,
+		ShowInNavigation: in.ShowInNavigation,
+		Order:            in.Order,
+		Settings: gqlschema.Settings{
+			ReadOnly: in.Settings.ReadOnly,
+		},
+	}
+
+	return &navigationNode, nil
+}
+
+func (c *clusterMicrofrontendConverter) NavigationNodesToGQLs(in []uiV1alpha1v.NavigationNode) ([]gqlschema.NavigationNode, error) {
```

Comment:
I think that this method should be private to package, because you only use it in methods `ToGQL` or `ToGQL`.